rq/group.py

```diff
@@ -0,0 +1,121 @@
+from typing import List, Optional, Union
+from uuid import uuid4
+
+from redis import Redis
+from redis.client import Pipeline
+
+from . import Queue
+from .exceptions import NoSuchGroupError
+from .job import Job
+from .queue import EnqueueData
+from .utils import as_text
+
+
+class Group:
+    """A Group is a container for tracking multiple jobs with a single identifier."""
+
+    REDIS_BATCH_NAME_PREFIX = 'rq:group:'
+    REDIS_BATCH_KEY = 'rq:groups'
+
+    def __init__(self, connection: Redis, id: str = None):
```

Comment:
Can we change `id` to `name`? `id` is a reserved word in Python, plus using `name` would give it more symmetry with `Queue(name='foo')`